k8s/operators/src/pool/diskpool/crd/v1beta3.rs

```diff
@@ -0,0 +1,244 @@
+    pub pool_status: Option<PoolStatus>,
+    /// Capacity as number of bytes.
+    pub capacity: u64,
+    /// Used number of bytes.
+    pub used: u64,
+    /// Available number of bytes.
+    pub available: u64,
+    /// Encryption enabled.
+    #[serde(default)]
+    pub encrypted: bool,
+}
+
+impl Default for DiskPoolStatus {
+    fn default() -> Self {
+        Self {
+            cr_state: CrPoolState::Creating,
+            pool_status: None,
+            capacity: 0,
+            used: 0,
+            available: 0,
+            encrypted: false,
+        }
+    }
+}
+
+impl DiskPoolStatus {
+    /// Set when Pool is not found for some reason.
+    pub fn not_found(status: &Option<Self>) -> Self {
+        Self {
+            cr_state: status.clone().unwrap_or_default().cr_state,
+            pool_status: None,
+            ..Default::default()
+        }
+    }
+
+    /// Set when operator is attempting to delete on pool.
+    #[cfg(feature = "openapi")]
+    pub fn terminating(p: Pool) -> Self {
+        let state = p.state.unwrap_or_default();
+        let free = if state.capacity > state.used {
+            state.capacity - state.used
+        } else {
+            0
+        };
+        Self {
+            cr_state: CrPoolState::Terminating,
+            pool_status: Some(state.status.into()),
+            capacity: state.capacity,
+            used: state.used,
+            available: free,
+            encrypted: state.encrypted,
+        }
+    }
+
+    /// Set when deleting a Pool which is not accessible.
+    pub fn terminating_when_unknown() -> Self {
+        Self {
+            cr_state: CrPoolState::Terminating,
+            pool_status: Some(PoolStatus::Unknown),
+            ..Default::default()
+        }
+    }
+
+    pub fn mark_unknown() -> Self {
+        Self {
+            cr_state: CrPoolState::Created,
+            pool_status: Some(PoolStatus::Unknown),
+            ..Default::default()
+        }
+    }
+}
+
+#[cfg(feature = "openapi")]
+impl From<RestPoolStatus> for PoolStatus {
+    fn from(p: RestPoolStatus) -> Self {
+        match p {
+            RestPoolStatus::Unknown => Self::Unknown,
+            RestPoolStatus::Online => Self::Online,
+            RestPoolStatus::Degraded => Self::Degraded,
+            RestPoolStatus::Faulted => Self::Faulted,
+        }
+    }
+}
+
+/// Returns DiskPoolStatus from Control plane pool object.
+#[cfg(feature = "openapi")]
+impl From<Pool> for DiskPoolStatus {
+    fn from(p: Pool) -> Self {
+        if let Some(state) = p.state {
+            let free = if state.capacity > state.used {
+                state.capacity - state.used
+            } else {
+                0
+            };
+            Self {
+                cr_state: CrPoolState::Created,
+                pool_status: Some(state.status.into()),
+                capacity: state.capacity,
+                used: state.used,
+                available: free,
+                encrypted: false,
```

Comment:
Should this come from state.encrypted or something?

Reply:
Opps, missed this.